### PR TITLE
Clarify when to use versionadded and versionchanged

### PR DESCRIPTION
**Pull request by erlend-aasland**
----
📚 Documentation preview 📚: https://cpython-devguide--1264.org.readthedocs.build/

### CODE REVIEW

**Review by hugovk — APPROVED** [review]
Thank you!

**Comment by erlend-aasland** [comment]
Thanks for the review; I'll wait for CAM to chime in before landing!

File: documentation/markup.rst [hunk 1]
```diff
@@ -1019,13 +1019,19 @@ units as well as normal text:
    feature, or a part of it, to the library or C API.  When this applies to an
    entire module, it should be placed at the top of the module section before
    any prose.
+   When a new attribute, function, or method is added, make sure a
+   ``versionadded`` directive is added.
```
Comment (JelleZijlstra) [inline documentation/markup.rst:1023]:
```suggestion
   When a new attribute, function, or method is added, a
   ``versionadded`` directive should be present at the end of its
   documentation block.
```
Also describe explicitly where to put the directive.

Comment (erlend-aasland) [reply documentation/markup.rst:1023]: That's an improvement! Thanks.

File: documentation/markup.rst [hunk 2]
```diff
@@ -1019,13 +1019,20 @@ units as well as normal text:
    feature, or a part of it, to the library or C API.  When this applies to an
    entire module, it should be placed at the top of the module section before
    any prose.
+   When a new attribute, function, or method is added, a
+   ``versionadded`` directive should be present at the end of its
+   documentation block.
```
Comment (CAM-Gerlach) [inline documentation/markup.rst:1024]: **Note**: _Requires adding a new ref target label, `.. _information-units`, above the eponymous section above.

```suggestion
   When adding a new API :ref:`with a directive <information-units>`
   (``class``, ``attribute``, ``function``, ``method``, ``c:type``, etc),
   a ``versionadded`` should be included at the end of its description block.
```

* More broadly construe the types of entities we're discussing here (not just attributes, functions and methods)
* Be more specific about which get `versionadded` (any object/API that is represented by a directive, vs. e.g. callable parameters, behaviors, etc. that are not).
* Crosslink the relevant section
* Use active voice
* Describe the location a little more precisely/less ambiguously

File: documentation/markup.rst [hunk 3]
```diff
@@ -1035,8 +1042,12 @@ units as well as normal text:
 
    Example::
 
-      .. versionchanged:: 3.1
-         The *spam* parameter was added.
+      .. function:: func(spam)
+
+         Return foo and bar with *spam* applied.
```
Comment (CAM-Gerlach) [inline documentation/markup.rst:1047]:
```suggestion
      .. function:: func(spam=False)

         Return foo and bar, optionally with *spam* applied.
```

Be a little more realistic and consistent with the previous (avoiding a compat break, which in this case would usually mean a new function).

**Review by CAM-Gerlach — COMMENTED** [review]
Thanks! A few suggestions, one of which requires adding a new ref target label, `.. _information-units`, above the eponymous section.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @erlend-aasland !

**Pull request merged**